rbc/irtools.py

```diff
@@ -338,31 +380,87 @@ def compile_to_LLVM(functions_and_signatures,
-                                          pipeline_class=pipeline_class)
+            try:
+                cres = compiler.compile_extra(typingctx=typing_context,
+                                              targetctx=target_context,
+                                              func=func,
+                                              args=args,
+                                              return_type=return_type,
+                                              flags=flags,
+                                              library=main_library,
+                                              locals={},
+                                              pipeline_class=pipeline_class)
+            except OmnisciUnsupportedError as msg:
+                for m in re.finditer(r'[|]OmnisciUnsupportedError[|](.*?)\n', str(msg), re.S):
+                    print(f'Skipping {fname} `{sig}`: {m.group(0)[25:]}')
+                continue
+            except nb_errors.TypingError as msg:
+                for m in re.finditer(r'[|]OmnisciUnsupportedError[|](.*?)\n', str(msg), re.S):
+                    print(f'Skipping {fname} `{sig}`: {m.group(0)[25:]}')
+                    break
+                else:
+                    raise
+                continue
+            except Exception:
+                raise
+
+            skip = False
+            intrinsics, declarations, defined, libraries = get_called_functions(
+                cres.library, cres.fndesc.llvm_func_name)
+            for f in declarations:
+                if target.is_gpu:
+                    if f.startswith('__nv_') and f[5:] in get_libdevice_functions():
+                        # numba supports libdevice functions starting from 0.52
+                        continue
+
+                if target.is_cpu:
+                    if f in get_libm_functions():
+                        # omniscidb is linked against m library
+                        continue
+                    if f in get_stdio_functions():
+                        # omniscidb is linked against stdio library
+                        continue
+                    if f in get_stdlib_functions():
+                        # omniscidb is linked against stdlib library
+                        continue
+
+                print(f'Skipping {fname} for {target.name} that uses unknown function `{f}`')
+                skip = True
+                break
+
+            for f in intrinsics:
+                if target.is_gpu:
+                    nf = get_llvm_name(f)
+                    if nf in get_nvvm_intrinsic_functions():
+                        continue
+                if target.is_cpu:
+                    nf = get_llvm_name(f)
+                    if nf in get_llvm_intrinsic_functions():
+                        continue
+                print(f'Skipping {fname} for {target.name} that uses unsupported intrinsic `{f}`')
+                skip = True
+                break
+
+            if skip:
+                continue
```

Comment:
Move this code to its own function to not blow `compile_to_LLVM` size.

Reply:
Done. The code is now in `compile_instance`.